optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-backend/src/main/java/org/optaplanner/workbench/screens/solver/backend/server/SolverValidator.java

```diff
@@ -188,7 +184,11 @@ private String resolveProjectName(final Path resourcePath) {
 
     private String getSolverConfigResource(final Path resourcePath,
                                            final KieModule kieWorkbenchModule) {
-        return Paths.convert(Paths.convert(resourcePath)).toURI().substring(Paths.convert(Paths.convert(kieWorkbenchModule.getRootPath())).toURI().length() + "/src/main/resources/".length());
+        final Path convertedResourcePath = Paths.convert(Paths.convert(resourcePath));
+        final Path convertedModuleRootPath = Paths.convert(Paths.convert(kieWorkbenchModule.getRootPath()));
+        final String result = convertedResourcePath.toURI().substring(getUriLength(convertedModuleRootPath) + "/src/main/resources/".length());
```

Comment:
@Rikkola I'd rather see taking advantage of existing API (FileSystem, Pahts) to join paths instead of string concatenation. There also might be a way to become independent on the presence of trailing "/" (`normalize()`?) without adding new logic in the form of `PathUtil`.

One more important thing: we should have a test for `getSolverConfigResource()` that demonstrates is works as expected. Ideally, if it had been there it would have failed after the upstream change that caused the bug.

Reply:
@yurloc The Paths.convert(Paths.convert(path)) here is really a stupid thing. The reason for this is that some times the start is "default://" and sometimes "git://" both still point to the same place. So yes, that could use something smarter.

Another issue is that the last character should not be "/". It also happens to be the file name, that makes no sense. The file name should be the folder. This only happens with git:// not file://.

Checking with @ederign later today about how to and where to fix this. Made a new ticket. There are few other Jiras that are caused by this same issue. https://issues.jboss.org/browse/AF-1034.

Reply:
Can we make use of `Path.normalize()` to get rid of the last "/"?
```
groovy> java.nio.file.Paths.get("/home/jlocker/").normalize() 
 Result: /home/jlocker
```